### ISSUE TITLE
[zeroc-ice] Using FindIce fails to find SLICE definitions

### ISSUE THREAD

**Issue by ethindp** [microsoft/vcpkg#30781]
**Describe the bug**
After installing zeroc-ice with vcpkg, the slice definitions cannot be found by CMake.

**Environment**
- OS: Windows
- Compiler: MSVC 19.34.31943

**To Reproduce**
1. `./vcpkg install zeroc-ice[*]`
2. Write a CMake project and include the following lines:
    ```cmake
    find_package(Ice COMPONENTS Ice++11 IceSSL++11 Slice REQUIRED)
    find_package(Ice COMPONENTS Glacier2++11 IceBox++11 IceDiscovery++11 IceGrid++11 IceLocatorDiscovery++11 IceStorm++11 IceUtil IceXML)
    target_link_libraries(your_target_here PRIVATE ${Ice_LIBRARIES})
    ```
3. Configure your project. CMake should give the following error: "Failed to find all Ice components (missing: Ice_SLICE_DIR)".

**Expected behavior**
The slice definitions should be found.

**Failure logs**
The logs aren't that long, so I'll just paste them inline:

```
-- The CXX compiler identification is MSVC 19.34.31943.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.34.31933/bin/HostX64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
[... finding of other packages...]
-- Ice version: 3.7.7
-- Found the following Ice libraries:
--   Ice++11 (required)
--   IceSSL++11 (required)
CMake Error at C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Failed to find all Ice components (missing: Ice_SLICE_DIR) (found version
  "3.7.7")
Call Stack (most recent call first):
  C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.25/Modules/FindIce.cmake:544 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  C:/Users/user/source/vcpkg/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  CMakeLists.txt:124 (find_package)


-- Configuring incomplete, errors occurred!
See also "[...]/build/CMakeFiles/CMakeOutput.log".
See also "[...]/build/CMakeFiles/CMakeError.log".
```

**Additional context**
None

**Comment by JonLiu1993** [comment]
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"
1> Working directory: C:\Users\test\source\repos\CMakeProject42\out\build\x64-Debug
1> [CMake] CMake Error at out/build/x64-Debug/CMakeFiles/3.20.21032501-MSVC_2/CMakeSystem.cmake:6 (include):
1> [CMake]   include could not find requested file:
1> [CMake] 
1> [CMake]     F:/Feature-test/node-api/vcpkg/scripts/buildsystems/vcpkg.cmake
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeLists.txt:6 (project)
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- Ice version: 
1> [CMake] -- The following Ice libraries were not found:
1> [CMake] --   Ice++11 (required)
1> [CMake] --   IceSSL++11 (required)
1> [CMake] --   Slice (required)
1> [CMake] CMake Error at C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
1> [CMake]   Failed to find all Ice components (missing: Ice_SLICE2CPP_EXECUTABLE
1> [CMake]   Ice_INCLUDE_DIR Ice_SLICE_DIR Ice_LIBRARY _Ice_REQUIRED_LIBS_FOUND)
1> [CMake] Call Stack (most recent call first):
1> [CMake]   C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
1> [CMake]   C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.20/Modules/FindIce.cmake:544 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
1> [CMake]   CMakeProject42/CMakeLists.txt:25 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> [CMake] See also "C:/Users/test/source/repos/CMakeProject42/out/build/x64-Debug/CMakeFiles/CMakeOutput.log".
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\ENTERPRISE\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 16 2019" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\CMakeProject42\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\CMakeProject42" 2>&1"' returned with exit code: 1'.

````
This issue could be reproduced with vcpkg 2023-04-07-bedcba5172f5e4b91caac660ab7afe92c27a9895 and VS2019 16.11.25.

**Comment by bold84** [comment]
I'll take care of it

**Comment by JonLiu1993** [comment]
@ethindp, I investigated some upstream ice3.7 version no longer provide Slice library, Slice library used to be provided as a shared library of Ice <3.7. As of Ice 3.7, the Slice compiler is now statically linked with the ice library, so it is no longer provided as a shared library, you should remove this from your project.
https://github.com/zeroc-ice/ice/issues/106#issuecomment-397612181

**Comment by JonLiu1993** [comment]
We hope your question was answered to your satisfaction; if it wasn't, you can reopen with more info.

**Comment by dg0yt** [comment]
Might be fixed with #33648.
